Challengue_22/Fanny_Gabriela_Becerra_Hernandez/main.js

```diff
@@ -0,0 +1,128 @@
+/***************************************************************************************************
+ * Name: ARRAYS_LoopsInJavascript
+ * Author:
+ * Description: This script tests for loops for an array.
+ * *************************************************************************************************
+ * LastModifiedBy     |     LastModifiedDate     |     Modification Id     |     Change description
+ * - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
+ * Fanny Becerra      |     26-Jun-2020          |     v0.1                |     Beta version
+ */
+
+let counterFor = 0,
+    counterForIn = 0,
+    counterForOf = 0,
+    counterWhile = 0,
+    fruitsArray = ["Aguacatito", "Berenjenita", "Pepinito feliz", "Tomatito bebé"],
+    index = 0,
+    table = null,
+    td = null,
+    tr = null,
+    txt = '';
+
+/**  @author:      Fanny Becerra.
+ *   @param:       None.
+ *   @return:      None.
+ *   @description: this function tests while loop.
+ */
+function funcWhile() {
+    table = document.getElementById('while');
+    console.log('WHILE');
+    if (counterWhile == 0) {
+        while (index < fruitsArray.length) {
+            console.log(fruitsArray[index]);
+            printRow(table, fruitsArray[index]);
+            index++;
+        }
+        index = 0;
+        counterWhile++;
+
+    } else {
+        deleteRows(table);
+        counterWhile = 0;
+    }
+}
+
+/**  @author:      Fanny Becerra.
+ *   @param:       None.
+ *   @return:      None.
+ *   @description: this function tests for loop.
+ */
+function funcFor() {
+    table = document.getElementById('for');
+    console.log('FOR');
+    if (counterFor == 0) {
+        for (index; index < fruitsArray.length; index++) {
```

Comment:
Falta declarar el index como let e incializarlo. Como buena prática de performance fruitsArray.length es mejor declararlo en una variable, ya que cada vez que entra al ciclo obtiene el tamaño del arreglo. En mismo sentido tiene mejor performance index += index que index++. El for refactorizado queda de la siguiente manera:
for(let index = 0, fruitsLength = fruitsArray.length; index += index)

Reply:
Inicialicé el index en la parte superior donde declaré todas las variables por primera vez. Como cada vez que entre a una función el index debe valer 0, para no declararlo dentro de cada función, lo hice al inicio. ¿Es mejor práctica inicializar la variable dentro de cada for en lugar de afuera en un scope más grande?